test/metabase/search/filter_test.clj

```diff
@@ -0,0 +1,58 @@
+(ns metabase.search.filter-test
+  (:require
+   [clojure.math.combinatorics :as math.combo]
+   [clojure.test :refer :all]
+   [metabase.search.config :as search.config]
+   [metabase.search.filter :as search.filter]
+   [metabase.search.in-place.filter :as search.in-place.filter]))
+
+(defn- filter-keys []
+  (keys @#'search.filter/context-key->attr))
+
+(defn- active-filter-combinations []
+  ;; We ignore :archived? as we've moved some of these filters to the :where clause as a simplifying optimization.
+  (math.combo/subsets (remove #{:archived?} (filter-keys))))
+
+(deftest search-context->applicable-models-test
+  (doseq [active-filters (active-filter-combinations)]
+    (testing (str "Consistent models included when filtering on " active-filters)
+      (let [search-ctx (assoc (zipmap active-filters (repeat true)) :models search.config/all-models)]
+        (is (= (search.in-place.filter/search-context->applicable-models search-ctx)
+               (search.filter/search-context->applicable-models search-ctx)))))))
+
+(def kitchen-sink-filter-context
+  {:archived?           true
+   :created-at          "2024-10-01"
+   :created-by          [123]
+   :last-edited-by      [321]
+   :last-edited-at      "2024-10-02"
+   :search-native-query true
```

Comment:
This doesn't affect the where-clause directly:

1. It restricts the models, but that happens via `search-context->applicable-models`.
2. It will otherwise *relax* the search to span the `dataset_query` field.